File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.php
class Solution {

    /**
     * @param Integer[] $nums1
     * @param Integer[] $nums2
     * @return Integer[]
     */
    function intersection($nums1, $nums2) {
        $counter = [];
        $intersection = [];

        foreach ($nums1 as $num) {
           $counter[$num] = $num;
        }

        foreach ($nums2 as $num) {
            if(isset($counter[$num]))
            {
                $intersection[] = $num;
                unset($counter[$num]);
            }
        }

        return $intersection;
    }
}

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.go
func intersection(nums1 []int, nums2 []int) []int {
    mapCnt := map[int]bool{}
    ans := []int{}
    
    for _, val := range nums1 {
        mapCnt[val] = true
    }
    
    for _, val := range nums2 {
        if mapCnt[val] == true {
            ans = append(ans, val)
            mapCnt[val] = false
        }
    }
    
    return ans
}